x-pack/plugins/reporting/export_types/csv_from_savedobject/server/lib/get_filters.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { badRequest } from 'boom';
+import moment from 'moment';
+
+import { SavedSearchObjectAttributes, TimeRangeParams } from '../../';
+import { QueryFilter, Filter, SearchSourceFilter } from './';
+
+export function getFilters(
+  indexPatternId: string,
+  indexPatternTimeField: string,
+  timerange: TimeRangeParams,
+  savedSearchObjectAttr: SavedSearchObjectAttributes,
+  searchSourceFilter: SearchSourceFilter,
+  queryFilter: QueryFilter
+) {
+  let includes: string[];
+  let timeFilter: any | null;
+  let timezone: string | null;
+
+  if (indexPatternTimeField) {
+    if (!timerange) {
+      throw badRequest(
+        `Time range params are required for index pattern [${indexPatternId}], using time field [${indexPatternTimeField}]`
+      );
+    }
+
+    const savedSearchCols = savedSearchObjectAttr.columns || [];
+    includes = [indexPatternTimeField, ...savedSearchCols];
+    timeFilter = {
+      range: {
+        [indexPatternTimeField]: {
+          format: 'epoch_millis',
+          gte: moment(timerange.min).valueOf(),
```

Comment:
Are we suspect of getting that negative value here? Not sure if ES gracefully handles this at this level